lib/matplotlib/figure.py

```diff
@@ -1776,8 +1776,13 @@ def get_tightbbox(self, renderer, bbox_extra_artists=None):
 
     @staticmethod
     def _normalize_grid_string(layout):
-        layout = inspect.cleandoc(layout)
-        return [list(ln) for ln in layout.strip('\n').split('\n')]
+        if '\n' not in layout:
```

Comment:
Why not do if ';' in layout? Should ';\nA' be valid?

Reply:
I've designed this PR to not change the behavior of multi-line strings. I've just *added* the possibility of single-line strings. Therefore the check is between those two options ("Do I have a line break?").

Whether or not we want stronger checking for the multi-line case is orthogonal.